flow/connectors/kafka/qrep.go

```diff
@@ -0,0 +1,111 @@
+package connkafka
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	lua "github.com/yuin/gopher-lua"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/model"
+	"github.com/PeerDB-io/peer-flow/pua"
+)
+
+func (*KafkaConnector) SetupQRepMetadataTables(_ context.Context, _ *protos.QRepConfig) error {
+	return nil
+}
+
+func (c *KafkaConnector) SyncQRepRecords(
+	ctx context.Context,
+	config *protos.QRepConfig,
+	partition *protos.QRepPartition,
+	stream *model.QRecordStream,
+) (int, error) {
+	startTime := time.Now()
+	numRecords := atomic.Int64{}
+	schema := stream.Schema()
+
+	wgCtx, wgErr := context.WithCancelCause(ctx)
+	pool, err := c.createPool(wgCtx, config.Script, config.FlowJobName, wgErr)
+	if err != nil {
+		return 0, err
+	}
+	defer pool.Close()
+
+Loop:
+	for {
+		select {
+		case qrecord, ok := <-stream.Records:
```

Comment:
seems like a lot of the code below is shared between pub-sub and Kafka, can we unify it?

Reply:
Also share quite a bit between cdc/qrep. It'll be a mess of conditions. Would be better as an isolated PR